endpoint/provider_specific_property_filter.go

```diff
@@ -0,0 +1,61 @@
+/*
```

Comment:
Could we move this instead to `pkg/filters` ? Example pull request https://github.com/gofogo/k8s-sigs-external-dns-fork/pull/3/files, is not officially a pull request against this repository, as there is another work that needs doing as well